sql-db/narayana-transactions/src/main/java/io.quarkus.ts.transactions/TransferTopUpService.java

```diff
@@ -0,0 +1,33 @@
+package io.quarkus.ts.transactions;
+
+import java.math.BigDecimal;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Named;
+
+import org.jboss.logging.Logger;
+
+import io.quarkus.narayana.jta.QuarkusTransaction;
+import io.quarkus.narayana.jta.RunOptions;
+
+@ApplicationScoped
+@Named("topup")
+public class TransferTopUpService extends TransferProcessor {
+    private static final Logger LOG = Logger.getLogger(TransferTopUpService.class);
+    private final static String ANNOTATION_TOP_UP = "user top up";
+    private final static int TRANSACTION_TIMEOUT_SEC = 10;
+
+    public JournalEntity makeTransaction(String from, String to, BigDecimal amount) {
+        LOG.infof("TopUp account %s amount %s", from, amount.toPlainString());
+        verifyAccounts(to);
```

Comment:
Is not really necessary because a toUp is just add money to your account. A Regular transaction is adding money to a different account. Let me add some comments on the resource methods in order to explain what each method does!